packages/connect-popup/src/index.tsx

```diff
@@ -181,6 +212,9 @@ const handleMessage = (event: MessageEvent<PopupEvent | UiEvent>) => {
 // handle POPUP.INIT message from window.opener
 const init = async (payload: PopupInit['payload']) => {
     if (!payload) return;
+
+    payload.systemInfo = getSystemInfo(config.supportedBrowsers);
```

Comment:
if (!payload.systemInfo) maybe lets do a fallback only for missing data?

Reply:
done and forcepushed